api/src/test/java/jakarta/data/metamodel/AttributeTest.java

```diff
@@ -89,7 +89,7 @@ void shouldCreateInRestriction() {
 
     @Test
     void shouldThrowExceptionForEmptyInRestriction() {
-        assertThatThrownBy(() -> testAttribute.in(Set.of()))
+        assertThatThrownBy(() -> testAttribute.in((Set<String>) null))
```

Comment:
A test that says it covers empty should actually pass in empty, not null,
```suggestion
        assertThatThrownBy(() -> testAttribute.in(Set.of()))
```